examples/protocol-and-stream-muxing/3.js

```diff
@@ -1,88 +1,72 @@
 /* eslint-disable no-console */
 'use strict'
 
-const libp2p = require('../../')
+const Libp2p = require('../../')
 const TCP = require('libp2p-tcp')
-const SPDY = require('libp2p-spdy')
+const MPLEX = require('libp2p-mplex')
+const SECIO = require('libp2p-secio')
 const PeerInfo = require('peer-info')
-const waterfall = require('async/waterfall')
-const parallel = require('async/parallel')
-const series = require('async/series')
-const pull = require('pull-stream')
-const defaultsDeep = require('@nodeutils/defaults-deep')
 
-class MyBundle extends libp2p {
-  constructor (_options) {
-    const defaults = {
-      modules: {
-        transport: [ TCP ],
-        streamMuxer: [ SPDY ]
-      }
-    }
-
-    super(defaultsDeep(_options, defaults))
-  }
-}
+const pipe = require('it-pipe')
+const { toBuffer } = require('it-buffer')
 
-function createNode (callback) {
-  let node
+const createNode = async () => {
+  const peerInfo = await PeerInfo.create()
+  peerInfo.multiaddrs.add('/ip4/0.0.0.0/tcp/0')
 
-  waterfall([
-    (cb) => PeerInfo.create(cb),
-    (peerInfo, cb) => {
-      peerInfo.multiaddrs.add('/ip4/0.0.0.0/tcp/0')
-      node = new MyBundle({
-        peerInfo
-      })
-      node.start(cb)
+  const node = await Libp2p.create({
+    peerInfo,
+    modules: {
+      transport: [TCP],
+      streamMuxer: [MPLEX],
+      connEncryption: [SECIO] // TODO: should not need this
     }
-  ], (err) => callback(err, node))
-}
+  })
 
-parallel([
-  (cb) => createNode(cb),
-  (cb) => createNode(cb)
-], (err, nodes) => {
-  if (err) { throw err }
+  await node.start()
 
-  const node1 = nodes[0]
-  const node2 = nodes[1]
+  return node
+}
 
-  node1.handle('/node-1', (protocol, conn) => {
-    pull(
-      conn,
-      pull.map((v) => v.toString()),
-      pull.log()
+;(async () => {
+  const [node1, node2] = await Promise.all([
+    createNode(),
+    createNode()
+  ])
+  
+  node1.handle('/node-1', ({ stream }) => {
+    pipe(
+      stream,
+      toBuffer,
+      source => (async function () {
+        for await (const msg of source) {
+          console.log(msg.toString())
+        }
+      })()
     )
   })
 
-  node2.handle('/node-2', (protocol, conn) => {
-    pull(
-      conn,
-      pull.map((v) => v.toString()),
-      pull.log()
+  node2.handle('/node-2', ({ stream }) => {
+    pipe(
+      stream,
+      toBuffer,
+      source => (async function () {
+        for await (const msg of source) {
+          console.log(msg.toString())
+        }
+      })()
```

Comment:
We can simplify the pipes in each of the example files as mentioned in the readme below.